### PR TITLE
style(ci): use standardized template

### PR DESCRIPTION
**Pull request by Fdawgs**
#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install Dependencies
+        run: |
+          npm install --ignore-scripts
+
+      - name: Check Licenses
+        run: |
+          npm run license-checker --if-present
```
[the remark this answers is not in the document]
Comment (Fdawgs) [reply .github/workflows/ci.yml]: Removed with https://github.com/fastify/fastify-basic-auth/pull/29/commits/79266ca35a168121ffa4f7acc2ff63da8c87699c

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
It seems that coveralls is configured by not working.

**Review by mcollina — CHANGES_REQUESTED** [review]
CI is failing because coveralls is not setup correctly or something.

**Comment by Fdawgs** [comment]
> 
> 
> CI is failing because coveralls is not setup correctly or something.

It looks okay and is using the same config as the main Fastify repo and other Fastify plugin repos.
I think it may be related to what you raised in https://github.com/fastify/fastify-accepts-serializer/pull/37#discussion_r590209007

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "types": "index.d.ts",
   "scripts": {
-    "test": "standard && tap test.js && tsd"
+    "test": "standard && tap test.js && tsd",
+    "test:ci": "standard && tap -J test.js --coverage-report=lcovonly && tsd"
```
Comment (mcollina) [inline package.json:9]: This change is not needed anymore.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install Dependencies
+        run: |
+          npm install --ignore-scripts
+
+      - name: Run Tests
+        run: |
+          npm run test:ci
```
Comment (mcollina) [inline .github/workflows/ci.yml:36]:
```suggestion
          npm run test
```

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by mcollina** [comment]
> > 
> 
> > 
> 
> > CI is failing because coveralls is not setup correctly or something.
> 
> 
> 
> It looks okay and is using the same config as the main Fastify repo and other Fastify plugin repos.
> 
> I think it may be related to what you raised in https://github.com/fastify/fastify-accepts-serializer/pull/37#discussion_r590209007

It's likely related. Enabling coveralls or analog for all repos requires significant work, so better avoid it entirely.

**Comment by mcollina** [comment]
The main Fastify repo is a special/ snowflake case.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**